cli/cmd/main.go

```diff
@@ -707,11 +794,64 @@ func main() {
 
 // DataCommand runs the data command for downloading data from the Viam cloud.
 func DataCommand(c *cli.Context) error {
-	if c.String(dataFlagDataType) != dataTypeBinary && c.String(dataFlagDataType) != dataTypeTabular {
+	dataType := c.String(dataFlagDataType)
+	if dataType != dataTypeBinary && dataType != dataTypeTabular {
+		return errors.Errorf("type must be binary or tabular, got %s", c.String("type"))
+	}
+
+	filter, err := createDataFilter(c)
+	if err != nil {
+		return err
+	}
+
+	client, err := rdkcli.NewAppClient(c)
+	if err != nil {
+		return err
+	}
+
+	if dataType == dataTypeBinary {
+		if err := client.BinaryData(c.String(dataFlagDestination), filter, c.Uint(dataFlagParallelDownloads)); err != nil {
+			return err
+		}
+	} else if dataType == dataTypeTabular {
+		if err := client.TabularData(c.String(dataFlagDestination), filter); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// DeleteCommand runs the command for deleting data from the Viam cloud.
+func DeleteCommand(c *cli.Context) error {
+	dataType := c.String(dataFlagDataType)
+	if dataType != dataTypeBinary && dataType != dataTypeTabular {
```

Comment:
I would still keep this as a switch case on `dataType` below in both command functions to avoid redundancy — we're doing a quick operation of creating a data filter and initializing an AppClient struct beforehand so ok wait to validate until after.

Reply:
make sense, done